### PR TITLE
Add basic go / pgx client tests

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: tests/client_tests/go/pg_type_workaround.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+-- dbext:type=CRATE:host=localhost:port=4200:dbname=doc
+
+
+create table doc.pg_type (
+    oid int,
+    typedelim string,
+    typelem int,
+    typname string,
+    typtype string
```
Comment (mxm) [inline tests/client_tests/go/pg_type_workaround.sql]: The naming seems odd `typedelim` vs `typname` but that's how it its.

File: tests/client_tests/go/basic_queries.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    "github.com/jackc/pgx"
+)
+
+
+func main() {
+    hosts := flag.String("hosts", "", "CrateDB hostname")
+    port := flag.Int("port", 5432, "CrateDB postgres port")
+    flag.Parse()
+
+	pgxConfig := pgx.ConnConfig{
```
Comment (mxm) [inline tests/client_tests/go/basic_queries.go]: indention off?

File: tests/client_tests/go/basic_queries.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		Host:     *hosts,
+		Port:     uint16(*port),
+		Database: "doc",
+		User:     "crate"}
+    conn, err := pgx.Connect(pgxConfig)
```
Comment (mxm) [inline tests/client_tests/go/basic_queries.go]: multiple return values, such a great language feature. Miss that from java :)

File: tests/client_tests/go/basic_queries.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    var name string
+    err = conn.QueryRow("select name || $1 from sys.cluster", "foo").Scan(&name)
+    if err != nil {
+        log.Fatal(err)
+        return
+    }
+    fmt.Println(name)
+    commandTag, err := conn.Exec("create table t1 (x int)")
+    if err != nil {
+        log.Fatal(err)
```
Comment (mxm) [inline tests/client_tests/go/basic_queries.go]: How are the results checked?

Comment (mfussenegger) [reply tests/client_tests/go/basic_queries.go]: If the statement causes an error it exists like this:

```
2017/12/01 13:33:54 cannot convert 2 to Text
exit status 1
Failure running: go run tests/client_tests/go/basic_queries.go --hosts 127.0.0.1 --port 5432
```

These tests are just simple "shouldn't run into an error" cases. The commandTag, results are tested in CrateDB itself.

File: tests/client_tests/go/pg_type_workaround.sql (hunk 1 above)
Comment (mfussenegger) [reply tests/client_tests/go/pg_type_workaround.sql]: oh this was actually a type - guess the typdelim column isn't used by pgx

File: tests/client_tests/go/basic_queries.go (hunk 2 above)
Comment (mfussenegger) [reply tests/client_tests/go/basic_queries.go]: ah yep I run that part through gofmt and didn't notice tabs vs. spaces difference in my editor =)

**Comment by mfussenegger** [comment]
@mxm fixup pushed

File: tests/client_tests/go/pg_type_workaround.sql (hunk 1 above)
Comment (mxm) [reply tests/client_tests/go/pg_type_workaround.sql]: What about `typelem` and `typtype`? they also have a different prefix.

Comment (mxm) [reply tests/client_tests/go/pg_type_workaround.sql]: Ah, the correct prefix is `typ`.

**Pull request merged**